tools/pipelines/build-docs-website-deploy.yml

```diff
@@ -0,0 +1,105 @@
+# Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+# Licensed under the MIT License.
+
+# build-docs-website-deploy pipeline
+# This pipeline triggers on the completion of the build-docs pipeline (main branch run).
+# After the build-docs pipeline completes, this pipeline downloads the generated fluidframework-docs
+# artifacts and deploys the content using AzureStaticWebApp.
+# This pipeline also has a scheduled trigger which runs every 4 hours.
+
+name: $(Build.BuildId)
+
+schedules:
+  - cron: "0 */4 * * *"
+    displayName: Website deploy with 4 hour increments
+    branches:
+      include:
+        - main
+    always: true
+pr: none
+
+parameters:
+- name: deployOverride
+  displayName: Deployment Override (default = based on branch)
+  type: string
+  default: default
+  values:
+    - default
+    - skip
+    - force
+- name: deployEnvironment
+  displayName: Static web app environment to deploy to
+  type: string
+  default: new
+  values:
+    - new
+    - old
+
+variables:
+- group: storage-vars
+- name: shouldDeploy
+  value: ${{ or(
+    eq(parameters.deployOverride, 'force'),
+    and(eq(variables['Build.SourceBranchName'], 'main'), eq(parameters.deployOverride, 'default'))
+    )}}
+- name: deploymentToken
+  ${{ if eq( parameters['deployEnvironment'], 'new' ) }}:
+    value: "$(FLUID_WEBSITE_TORUS_API_TOKEN)"
+  ${{ if eq( parameters['deployEnvironment'], 'old') }}:
+    value: "$(AZURE_STATIC_WEB_APPS_API_TOKEN)"
+
+# We are setting up a pipeline resource that references the build-docs
+# pipeline and setting up a pipeline completion trigger so that our build-docs-website-deploy
+# pipeline runs when a run of the build-docs pipeline completes
+resources:
+  pipelines:
+  - pipeline: build-docs # Name of the pipeline resource.
+    source: build-docs # The name of the pipeline referenced by this pipeline resource.
+    trigger: 
+      branches:
+        include: 
+        - main # Run pipeline when build-docs main branch pipeline completes
```

Comment:
I _think_ this wants to trigger off of `build-docs` running on main or on release branches, but _this_ pipeline should only run on main. It isn't clear to me which scenario this branch filter is for (the branch the triggering pipeline was run on, or the branch this pipeline should run on), but I think that's what we ultimately want. And in that case, I don't think we need the daily trigger after all, since we have the exact fine-grained control we want.

Reply:
The trigger branches specified here are referring to the branch run on the `build-docs` pipeline. However, looking at the "branch considerations" section of this document, it seems like the second pipeline will use the same branch if the two branches are in the same repository. https://learn.microsoft.com/en-us/azure/devops/pipelines/process/pipeline-triggers?view=azure-devops

Reply:
I'm a bit confused by this triggering on changes to main. We could have documentation for unreleased APIs there, right? So we wouldn't want those to make it to the public site?

Reply:
That's true. Long term, we still want to publish a "latest", or "upcoming" docs artifact from `main` to power local website builds (to enable previewing upcoming contents). But it's true that we don't want to publish that to the public website. So I guess we just want to trigger off of `releases` branch runs of `build-docs` and run this on `main`.